### PR TITLE
docs: new renovate flow graph

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Changes

added new renovate flow picture

## Context

Outdated renovate flow, not enough info for developers.

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Are there any existing docs you think it's worth linking from or embedding within?

Does it replace the old one?

I don't think putting 2022 in the file name is useful - git has timestamps already :)

**Comment by HonkingGoose** [comment]
GitHub can show a `Mermaid` diagram in Markdown files. Read the [Github Blog, Include diagrams in your Markdown files with Mermaid](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) to learn more.

A Mermaid diagram is text-based, so that _might_ be easier to update later, as we won't have to create a new graph with a image editor. But editing the Mermaid code is tricky as well. 🙈 😄 

I'll let the maintainers decide. 😉

**Comment by HonkingGoose** [comment]
> Does it replace the old one?

Here's the old diagram:

https://github.com/renovatebot/renovate/blob/main/docs/development/assets/renovate-logic-flow.jpg

> Are there any existing docs you think it's worth linking from or embedding within?

We could link to the diagram from our [design decisions](https://github.com/renovatebot/renovate/blob/main/docs/development/design-decisions.md) document, maybe.

File: docs/development/assets/RenovateFlow.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    BC --> BD[do all validation]
+    BC --> BF[schedules]
+    BC --> BG[updates]
+    BC --> BH[ensurePR]
+    end
+
+    Start ---> Repository
+    K ---> initializeRepository
+    L ----> extractDependencies
+    N -----> updateRepository
```
[the remark this answers is not in the document]
Comment (PhilipAbed) [reply docs/development/assets/RenovateFlow.md]: are u serious??? :D i directly checked the github integration as my webstorm/vscode couldnt read it so i thought i need to compile it somehow

**Comment by PhilipAbed** [comment]
results are here: https://github.com/PhilipAbed/renovate/blob/renovateFlow/docs/development/assets/renovate-flow.md

File: docs/development/assets/RenovateFlow.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (HonkingGoose) [reply docs/development/assets/RenovateFlow.md]: Yeah I'm serious! 🙃 

If you edit your Mermaid diagram from within the GitHub website then you can always switch to the file preview screen, and see how the diagram looks. No need to compile then.

We'll need to decide if we want to rely on people viewing the files on GitHub though. If we want to publish the files outside of GitHub we'll still need to convert the Mermaid code to a proper picture.

**Comment by rarkins** [comment]
Looks great!

**Comment by PhilipAbed** [comment]
there's another option to make it Top down, like this:
image

**Comment by PhilipAbed** [comment]
@HonkingGoose let me know which option looks better, the picture, or the one you mentioned before 

also please advise if i'm missing something, or if you guys want to add anything else to it.

i tried to show the full picture

**Comment by HonkingGoose** [comment]
> @HonkingGoose let me know which option looks better, the picture, or the one you mentioned before

The "Top Down" view is harder to follow. I'm happy with what we merged. 🥳

**Pull request merged**